src/api/channels.tsx

```diff
@@ -1,29 +1,52 @@
 import React from 'react';
-import { AxiosError } from 'axios';
+import axios, { AxiosError } from 'axios';
+import { Channel, FullChannelMetadata, Record } from '../app.types';
 import { useQuery, UseQueryResult, UseQueryOptions } from 'react-query';
 import { Column } from 'react-table';
-import { FullChannelMetadata } from '../app.types';
-import { getFullChannelMetadata, randomNumber } from '../recordGeneration';
 import { roundNumber } from '../table/cellRenderers/cellContentRenderers';
+import { selectUrls } from '../state/slices/configSlice';
+import { useAppSelector } from '../state/hooks';
 
-const sleep = (ms: number): Promise<unknown> => {
-  return new Promise((resolve) => setTimeout(resolve, ms));
+export const generateChannelMetadata = (
+  records: Record[]
+): FullChannelMetadata[] => {
+  let metadata: FullChannelMetadata[] = [];
+
+  records.forEach((record: Record) => {
+    const keys = Object.keys(record.channels);
+    keys.forEach((key: string) => {
+      if (!metadata.find((channel) => channel.systemName === key)) {
+        const channel: Channel = record.channels[key];
+        const channelDataType = channel.metadata.channel_dtype;
+        const newMetadata: FullChannelMetadata = {
+          systemName: key,
+          channel_dtype: channelDataType,
+        };
+        metadata.push(newMetadata);
+      }
+    });
+  });
+
+  return metadata;
 };
 
-// TODO change this when we have an API to query
```

Comment:
Can we add back this `TODO` to remind us that this is still a WIP way to fetch channels?